### ISSUE TITLE
GHC 8.4 semigroup/monoid change

### ISSUE THREAD

**Issue by clinty** [blamario/incremental-parser#3]
```
Text/ParserCombinators/Incremental.hs:177:10: error:
    • Could not deduce (Semigroup (Parser t s r))
        arising from the superclasses of an instance declaration
      from the context: (Monoid s, Monoid r)
        bound by the instance declaration
        at Text/ParserCombinators/Incremental.hs:177:10-54
    • In the instance declaration for ‘Monoid (Parser t s r)’
    |
177 | instance (Monoid s, Monoid r) => Monoid (Parser t s r) where
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by blamario** [comment]
Fixed by ce30037c53042fde7a51cefb1d989e306e896a4f in version 0.3

**Comment by clinty** [comment]
Yeah, that works, but incremental-parser 0.3 is broken with GHC 8.0 and GHC 8.2

Thanks though
